packages/core/src/new-api/internal/reconciliation/reconcilier.ts

```diff
@@ -0,0 +1,115 @@
+import { ExecutionState, ExecutionStateMap } from "../../types/execution-state";
+import { Future, IgnitionModule, ModuleParameters } from "../../types/module";
+import { AdjacencyList } from "../utils/adjacency-list";
+import { AdjacencyListConverter } from "../utils/adjacency-list-converter";
+
+import { reconcileCurrentAndPreviousTypeMatch } from "./reconcileCurrentAndPreviousTypeMatch";
+import { reconcileDependencyRules } from "./reconcileDependencyRules";
+import { reconcileFutureSpecificReconciliations } from "./reconcileFutureSpecificReconciliations";
+import {
+  ReconciliationCheck,
+  ReconciliationContext,
+  ReconciliationFailure,
+  ReconciliationFutureResult,
+  ReconciliationFutureResultFailure,
+  ReconciliationResult,
+} from "./types";
+
+export class Reconcilier {
+  public static reconcile(
+    module: IgnitionModule,
+    executionStateMap: ExecutionStateMap,
+    moduleParameters: ModuleParameters,
```

Comment:
I think this should be a `Map` from module id to `ModuleParameters`. I believe this got lost in the PR that introduced runtime values.